source/postprocess/particles.cc

```diff
@@ -354,141 +354,147 @@ namespace aspect
                                                this->get_time() / year_in_seconds :
                                                this->get_time());
 
-      if (output_format=="none")
+      for (std::vector<std::string>::iterator itr_output_format = output_formats.begin();
+           itr_output_format != output_formats.end();
+           itr_output_format++)
         {
-          // If we do not write output return early with the number of advected particles
-          return std::make_pair("Number of advected particles:",
-                                Utilities::int_to_string(world.n_global_particles()));
-        }
-      else if (output_format=="hdf5")
-        {
-          const std::string particle_file_name = "particles/" + particle_file_prefix + ".h5";
-          const std::string xdmf_filename = "particles.xdmf";
-
-          // Do not filter redundant values, there are no duplicate particles
-          DataOutBase::DataOutFilter data_filter(DataOutBase::DataOutFilterFlags(false, true));
-
-          data_out.write_filtered_data(data_filter);
-          data_out.write_hdf5_parallel(data_filter,
-                                       this->get_output_directory()+particle_file_name,
-                                       this->get_mpi_communicator());
-
-          const XDMFEntry new_xdmf_entry = data_out.create_xdmf_entry(data_filter,
-                                                                      particle_file_name,
-                                                                      time_in_years_or_seconds,
-                                                                      this->get_mpi_communicator());
-          xdmf_entries.push_back(new_xdmf_entry);
-          data_out.write_xdmf_file(xdmf_entries, this->get_output_directory() + xdmf_filename,
-                                   this->get_mpi_communicator());
-        }
-      else if (output_format=="vtu")
-        {
-          // Write master files (.pvtu,.pvd,.visit) on the master process
-          const int my_id = Utilities::MPI::this_mpi_process(this->get_mpi_communicator());
-
-          if (my_id == 0)
+          std::string output_format = *itr_output_format;
```

Comment:
Why do you need to make this copy? Would `*output_format == "none" in the next line not work?
Or is this to preserve the rest of the code? 

I would prefer one of two changes:
- Either use the iterator instead of making the copy
- or use a loop over `i` and always use `output_formats[i]`